subsys/net/lib/nrf_provisioning/src/nrf_provisioning_coap.c

```diff
@@ -571,6 +604,11 @@ int nrf_provisioning_coap_req(struct nrf_provisioning_coap_context *const coap_c
 			cdc_ctx.opkt = tx_buf.coap;
 			cdc_ctx.opkt_sz = sizeof(tx_buf);
 
+			if (socket_keep_open) {
+				LOG_DBG("Requesting DTLS session suspension");
+				suspend_dtls_session(&client, coap_ctx);
```

Comment:
If this fails, would it be better to fall back on closing the socket?

Reply:
Good point. Now socket_keep_open is set to false if DTLS session suspension fails.